resolvers.py

```diff
@@ -0,0 +1,1979 @@
+
+        return nested_data
+
+
+class NestingOfReleasesFacts(NestingHelper):
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        # Adding Master Filter for billability and Multiple Account Names
+        billability = kwargs.get("billability")
+        filters = kwargs.get("filters")
+        flat_data = models.ReleasesFacts.objects.all()
+        flat_data = flat_data.filter(billability=billability)
+        if filters:
+            flat_data = flat_data.filter(account_name__in=filters)
+
+        # Making a Intermediate Grouped Data for our required Nested API
+        grouped_data = self.group_data(flat_data)
+
+        # Nested API
+        nested_data = []
+        for account_tuple, account_data in grouped_data.items():
+            nested_data.append(self.process_account_data(
+                account_tuple, account_data))
+
+        return nested_data
+
+
+class NewAllocationAndReleaseAllocatedHasherCellDownload(AllocatedHasherHelper):
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+
+        pod_id = kwargs.get("pod_id")
+        project_name = kwargs.get("project_name")
+        account_name = kwargs.get("account_name")
+        billability = kwargs.get("billability")
+        duration = kwargs.get('duration').strip()
+        report_name = kwargs.get('report_name').strip().lower()
+        start_date = kwargs.get('start_date')
+        end_date = kwargs.get('end_date')
+
+        return self.cell_download_helper(pod_id, project_name, account_name, billability, duration, start_date, end_date, report_name)
+
+
+class OverallFreePoolByBandFilteredSummary():
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        # filters variable takes band fields than needs to be shown
+        filters = kwargs.get("filters")
+        qs = []
+        if (len(filters) == 0):
+            qs = models.OverallBenchByBand.objects.all()
+        else:
+            qs_filtered = models.OverallBenchByBand.objects.filter(
+                band_name__in=filters)
+            qs_sum = qs_filtered.aggregate(bench=Sum('bench'), lop=Sum('lop'), resigned=Sum('resigned'), maternity=Sum('maternity'), allocation_pending=Sum('allocation_pending'), nominated=Sum('nominated'), non_deployable_total=Sum(
+                'non_deployable_total'), deployable_total=Sum('deployable_total'), sabbatical=Sum('sabbatical'), grand_total=Sum('grand_total'), grand_total_percentage=Sum('grand_total_percentage'))
+            qs_dummy = {"id": "Total", "band_name": 'Total'}
+            qs_dummy.update(qs_sum)
+            qs = list(chain(qs_filtered, [qs_dummy]))
+        return qs
+
+
+class OverallFreePoolByCapabilityFilteredSummary():
+    @custom_resolver_check_perm
+    def __call__(self, source, info, **kwargs):
+        # filters variable takes band fields than needs to be shown
+        filters = kwargs.get("filters")
+        qs = []
+        if (len(filters) == 0):
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)